src/secure_random.cr

```diff
@@ -19,4 +29,16 @@ module SecureRandom
     end
     slice
   end
+
+  def self.uuid
+    bytes = random_bytes(16)
+    {
+      bytes[0, 4].hexstring,
+      bytes[4, 2].hexstring,
+      bytes[6, 2].hexstring,
+      bytes[8, 2].hexstring,
+      bytes[10, 2].hexstring,
+      bytes[12, 4].hexstring,
+    }.join("-")
```

Comment:
I tried this:

```crystal
require "benchmark"

TUP = {"a", "b", "c", "d", "e", "f", "g"}

Benchmark.ips do |x|
  x.report("string") do
    TUP.join "-"
  end
  x.report("char") do
    TUP.join '-'
  end
end
```

and get:

```
string 3476001.50 (± 23731.54)  1.17× slower
  char 4076697.63 (± 27724.80)       fastest
```

In any case you are generating many intermediate strings there (the `hexstring` calls). The best thing would be to have a separate overload for `hexstring` accepting an IO, then using Enumerable#join with IO and a block.